cmd/kube-scheduler/app/options/configfile.go

```diff
@@ -76,7 +78,7 @@ func WriteConfigFile(fileName string, cfg *kubeschedulerconfig.KubeSchedulerConf
 		return fmt.Errorf("unable to locate encoder -- %q is not a supported media type", mediaType)
 	}
 
-	encoder := kubeschedulerscheme.Codecs.EncoderForVersion(info.Serializer, kubeschedulerconfigv1alpha1.SchemeGroupVersion)
+	encoder := kubeschedulerscheme.Codecs.EncoderForVersion(info.Serializer, kubeschedulerconfigv1alpha2.SchemeGroupVersion)
```

Comment:
I don't know when this is invoked but this doesn't look rollback safe.

Reply:
This is only used for debugging purposes.
If you specify an output config file, we produce it and then exit the binary. This way a dev can verify defaults and/or validation is applied properly.
So, this should be safe, as everything will be converted to v1alpha2 internally.